### PR TITLE
revng ptml: merge cat and strip

### PR DESCRIPTION
**Pull request by mrjackv**
Merge the cat and strip subcommands within `revng ptml` Refactor codebase so that both color and plaintext output are handled in a unified manner.

### CODE REVIEW

File: python/revng/cli/_commands/ptml/text.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        output = args.input
+    else:
+        output = args.output
+
+    if not args.color:
+        console = PlainConsole(output)
+    else:
+        try:
+            # rich takes a while to import, do it if needed
+            from rich.console import Console
```
Comment (aleclearmind) [inline python/revng/cli/_commands/ptml/text.py:101]: Do we have a PR for adding this as a dep?

File: python/revng/cli/_commands/ptml/text.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+            console = Console(markup=False, highlight=False, force_terminal=True, file=output)
+            console.options.no_wrap = True
+            color = True
+        except ImportError:
+            console = PlainConsole(output)
+            color = False
+
+    if args.color and not color:
+        print("Module 'rich' not found, please install it to use color mode")
```
Comment (aleclearmind) [inline python/revng/cli/_commands/ptml/text.py:111]:
```suggestion
        log("Module 'rich' not found, please install it to use color mode")
```

File: python/revng/cli/_commands/ptml/text.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                console.print(content, end="", style=COLOR_CONVERSION[metadata["data-token"]])
+            else:
+                console.print(content, end="")
+        elif node.nodeType == Node.ELEMENT_NODE:
+            new_metadata = {**metadata}
+            for key, value in node.attributes.items():
+                new_metadata[key] = value
+            _parse_ptml_node(node, console, indent, new_metadata)
+        else:
+            pass
```
Comment (aleclearmind) [inline python/revng/cli/_commands/ptml/text.py:78]: ?
```suggestion
```

File: python/revng/cli/_commands/ptml/text.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+def parse_ptml_plain(content: str, console):
+    dom = parseString(content)
+    _parse_ptml_node(dom, console, "", {})
+
+
+def parse_ptml_yaml(content: Dict[str, str], console):
+    for key, value in content.items():
+        console.print(f"{key}:", style="yellow1", end="")
+        console.print(" |-")
```
Comment (aleclearmind) [inline python/revng/cli/_commands/ptml/text.py:57]: Are you sure that using `|-` you can dump whatever you want and things works just by prefixing with `  `?
I'd feel safer if we'd just modify the document in place and use `yaml.dump`.

File: python/revng/cli/_commands/ptml/text.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    if args.color and not color:
+        print("Module 'rich' not found, please install it to use color mode")
+        sys.exit(1)
+
+    handle_file(
+        content,
+        lambda x: parse_ptml_plain(x, console),
+        lambda x: parse_ptml_yaml(x, console),
+        filters,
+    )
```
Comment (aleclearmind) [inline python/revng/cli/_commands/ptml/text.py:119]: Handle file returns a value that you seem to be ignoring.
If I'm looking at the wrong function... consider giving different names, it'd be a `make_manager` situation.

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: python/revng/cli/_commands/ptml/text.py (hunk 1 above)
Comment (mrjackv) [reply python/revng/cli/_commands/ptml/text.py:101]: This was already added in the first `revng ptml` PR
https://github.com/revng/orchestra/commit/8378db3c539f1fbd482f2f0613ada8342e724ee4

File: python/revng/cli/_commands/ptml/text.py (hunk 4 above)
Comment (mrjackv) [reply python/revng/cli/_commands/ptml/text.py:57]: There are 4 main issues:
1. By using yaml.dump the keys can't be colored as easiliy
2. Currently pyyaml needs a [slight hack](https://github.com/yaml/pyyaml/issues/121#issuecomment-1018117110) to make flow scalars work
3. `rich` works mainly on the terminal, to make it work with yaml I'd have to have a fake file-like object where rich prints in it and then repackage it in a dictionary, which is inefficient and would bloat the codebase
4. If the situation was more generic (i.e. we don't know ahead of time the structure of the output) I'd 100% agree with using `yaml.dump`, however the structure of the output is fixed, so there (as far as I see) no situations where the output would get broken

File: python/revng/cli/_commands/ptml/text.py (hunk 2 above)
Comment (mrjackv) [reply python/revng/cli/_commands/ptml/text.py:111]: Fixed

File: python/revng/cli/_commands/ptml/text.py (hunk 3 above)
Comment (mrjackv) [reply python/revng/cli/_commands/ptml/text.py]: Dropped

File: python/revng/cli/_commands/ptml/text.py (hunk 5 above)
Comment (mrjackv) [reply python/revng/cli/_commands/ptml/text.py]: Fixed, `handle_file` is mean to forward the exit code from the `parse_*` function

File: python/revng/cli/_commands/ptml/text.py (hunk 4 above)
Comment (aleclearmind) [reply python/revng/cli/_commands/ptml/text.py:57]: OK, please add a comment about these reasons.

Comment (mrjackv) [reply python/revng/cli/_commands/ptml/text.py:57]: Added

**Comment by aleclearmind** [comment]
Please rebase

**Comment by mrjackv** [comment]
Rebased

**Comment by aleclearmind** [comment]
Merged in df68b2c.

**Pull request closed without merging**